### ISSUE TITLE
git repo tweaks for db/schema.rb and workflowservice.log

### ISSUE THREAD

**Issue by darrenleeweber** [sul-dlss/argo#59]
On a 'clean' checkout of the develop branch, after running the specs, the git repo isn't clean, i.e.

```
$ git status
On branch develop
Your branch is up-to-date with 'origin/develop'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

    modified:   db/schema.rb

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    workflow_service.log

no changes added to commit (use "git add" and/or "git commit -a")
```

1. Should `db/schema.rb` be in the .gitignore file? If not, should changes be committed?
2. Can the config settings be changed to put the `workflow_service.log` into a log path or somewhere that is already in .gitignore?

**Comment by atz** [comment]
Can gitignore `workflow.log*`.  I thought I already had done that.  Must've been in dor-services.

**Comment by atz** [comment]
`db/schema.rb` is still required for db:migrate, afaict.  What are the `schema.rb` diffs?

**Comment by jmartin-sul** [comment]
yeah, we should gitignore workflow.log* (doesn't appear to be ignored at the moment in argo).

db/schema.db shouldn't change when running the tests.  that's odd, though vaguely familiar.

instead of using `rake spec` to run the tests, and `rake server` to start the instance, try `rspec` and `rails server` (actually, i tend to run `rails s -u`, so that it's easy to insert `binding.pry` statements into the code for debugging).  it seems possible that by using rake to invoke tests or start the server, you're doing build stuff, and it's doing some sort of schema migration that it shouldn't be doing.

i'd also be curious at the diff in schema.rb.  maybe it's something to do with sqlite vs mysql?  the diffs i've seen in the past have been around thing like explicitly stating length limits on columns.  but i don't think any of that's ever been formally documented anywhere, just trouble-shot as needed when people were setting up laptop instances (e.g., i recall something similar when @tingulfsen was getting set up).

**Comment by mejackreed** [comment]
`schema.rb` diffs are due to running sqlite vs mysql [diff](https://github.com/sul-dlss/argo/commit/933e70a16b10032be732a7ba184ae24c0377e7be#diff-1acd2e7e27a227829d5d14a91c863bb6) with btree

this should be vanilla as possible using sqlite. Though we run MySQL in prod, we don't see any issues with this setup in all of our other Blacklight apps.

Ref: 
 - https://github.com/sul-dlss/SearchWorks/blob/master/db/schema.rb
 - https://github.com/sul-dlss/earthworks/blob/master/db/schema.rb

TLDR: Specs and dev using `rake db:migrate` shouldn't change this file.

**Comment by darrenleeweber** [comment]
```
 git diff db/schema.rb
diff --git a/db/schema.rb b/db/schema.rb
index 3412b2f..51c4651 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -23,7 +23,7 @@ ActiveRecord::Schema.define(version: 20150619200031) do
     t.string   "document_type", limit: 255
   end
 
-  add_index "bookmarks", ["user_id"], name: "index_bookmarks_on_user_id", using: :btree
+  add_index "bookmarks", ["user_id"], name: "index_bookmarks_on_user_id"
 
   create_table "delayed_jobs", force: :cascade do |t|
     t.integer  "priority",   limit: 4,     default: 0, null: false
@@ -39,7 +39,7 @@ ActiveRecord::Schema.define(version: 20150619200031) do
     t.datetime "updated_at"
   end
 
-  add_index "delayed_jobs", ["priority", "run_at"], name: "delayed_jobs_priority", using: :btree
+  add_index "delayed_jobs", ["priority", "run_at"], name: "delayed_jobs_priority"
 
   create_table "indexing_exceptions", force: :cascade do |t|
     t.string   "pid",                  limit: 255
@@ -50,7 +50,7 @@ ActiveRecord::Schema.define(version: 20150619200031) do
     t.datetime "updated_at"
   end
 
-  add_index "indexing_exceptions", ["pid"], name: "index_indexing_exceptions_on_pid", using: :btree
+  add_index "indexing_exceptions", ["pid"], name: "index_indexing_exceptions_on_pid"
 
   create_table "robots", force: :cascade do |t|
     t.string   "wf",         limit: 255
@@ -67,7 +67,7 @@ ActiveRecord::Schema.define(version: 20150619200031) do
     t.string   "user_type",    limit: 255
   end
 
-  add_index "searches", ["user_id"], name: "index_searches_on_user_id", using: :btree
+  add_index "searches", ["user_id"], name: "index_searches_on_user_id"
 
   create_table "users", force: :cascade do |t|
     t.string   "sunetid",    limit: 255
```

**Comment by darrenleeweber** [comment]
The diff is gone after `git checkout db/schema.rb` and it remains 'gone' after both `rspec` and `rake spec`, so the diff must have been generated by something else in the setup instructions.  So the candidates are:
```
rails generate argo:solr
rake argo:jetty:clean
rake argo:jetty:config
rake db:setup
rake db:migrate
rake tmp:create
```
The winner is.....  `rake db/migrate`!  Congratulations, your the proud winner of modifications to `db/schema.rb` and the committee is out on whether your changes will be accepted or not.  Have a nice day!

**Comment by atz** [comment]
`db:migrate` is straight from ActiveSupport.  The only task that should ever modify schema.rb is `db:schema:dump`.

**Comment by mejackreed** [comment]
Think this was an artifact of getting setup... Good to close?
